core/object/method_bind.h

```diff
@@ -51,14 +51,21 @@ class MethodBind {
 	bool _returns_raw_obj_ptr = false;
 
 protected:
-	Variant::Type *argument_types = nullptr;
+	bool _argument_types_dynamic_allocated = true;
+	// {Return Type, Arg1 Type, Arg2 Type, ..., ArgN Type}
+	const Variant::Type *argument_types = nullptr;
 #ifdef DEBUG_ENABLED
 	Vector<StringName> arg_names;
 #endif // DEBUG_ENABLED
 	void _set_const(bool p_const);
 	void _set_static(bool p_static);
 	void _set_returns(bool p_returns);
-	virtual Variant::Type _gen_argument_type(int p_arg) const = 0;
+	virtual Variant::Type _gen_argument_type(int p_arg) const {
```

Comment:
This should not be `virtual`, and can be renamed since nothing is 'generated'.

```suggestion
	Variant::Type _get_argument_type(int p_arg) const {
```

Reply:
`GDExtensionMethodBind` and `MethodBindVarArg` still need to override this function, so we cannot remove this virtual qualifier. Rename is possible.